custom_components/waste_collection_schedule/waste_collection_schedule/source/kiertokapula_fi.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from datetime import datetime
+
+import requests
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "Kiertokapula Finland"
+DESCRIPTION = "Schedule for kiertokapula FI"
+URL = "https://www.kiertokapula.fi"
+TEST_CASES = {
+    # No test cases, because auth :(
+}
```

Comment:
You can add test cases using 
```python
{
"bill_number": "!secret kiertonkapula_fi_bill_number"
"password": "!secret kiertonkapula_fi_bill_password"
}
```
and define these secrets in the custom_components/waste_collection_schedule/waste_collection_schedule/test/secrets.yaml file (which is not synchronized with git):

```yaml
kiertonkapula_fi_bill_number: NUMBER
kiertonkapula_fi_bill_password: PASSWORD
```